js/src/redux/providers/certifications/reducer.js

```diff
@@ -17,17 +17,25 @@
 const initialState = {};
 
 export default (state = initialState, action) => {
-  if (action.type !== 'addCertification') {
-    return state;
+  if (action.type === 'addCertification') {
+    const { address, id, name, icon, title } = action;
+    const certifications = state[address] || [];
+
+    if (certifications.some((c) => c.id === id)) {
+      return state;
+    }
+    const newCertifications = certifications.concat({ id, name, icon, title });
```

Comment:
linebreak before would be good, the rest of the callback is actually nice and readable.